onnxruntime/core/providers/xnnpack/xnnpack_execution_provider.cc

```diff
@@ -87,14 +88,38 @@ std::unique_ptr<KernelRegistry> RegisterKernels() {
 using namespace xnnpack;
 
 XnnpackExecutionProvider::XnnpackExecutionProvider(const XnnpackExecutionProviderInfo& info)
-    : IExecutionProvider{kXnnpackExecutionProvider, true} {
+    : IExecutionProvider{kXnnpackExecutionProvider, true}, info_(info) {
   if (info.xnn_thread_pool_size > 1) {
     // pthreadpool is independent of ort-threadpoool, so we have to disable cpu spinning for ort-threadpool.
     // otherwise, the pthreadpool will be starved and harm performance a lot.
     xnnpack_thread_pool_ = pthreadpool_create(static_cast<size_t>(info.xnn_thread_pool_size));
   }
 }
 
+common::Status XnnpackExecutionProvider::CheckSessionOptions(const SessionOptions& so) {
+  if (so.execution_mode != ExecutionMode::ORT_SEQUENTIAL) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
+                           "Parallel execution mode does not support the Xnnpack Execution Provider. "
+                           "Please set the execution mode as sequential for this "
+                           "session since it uses the Xnnpack Execution Provider.");
+  }
+  if (so.intra_op_param.allow_spinning && so.intra_op_param.thread_pool_size > 1) {
+    LOGS_DEFAULT(WARNING)
+        << "XNNPACK EP utilize pthreadpool for multi-threading. So, if allow_spinning on ORT's"
```

Comment:
suggestion:
The XNNPACK EP utilizes an internal pthread-based thread pool for multi-threading. 
If ORT's thread pool size is > 1 and spinning is enabled there will be contention between the two thread pools, and performance will suffer.
Please set either intra_op_param.allow_spinning to 0 in the SessionOption config params, or the ORT intra-op threadpool size to 1 and try again.